improver/cli/wind_gust_diagnostic.py

```diff
@@ -64,9 +64,14 @@ def process(
         iris.cube.Cube:
             Cube containing the wind-gust diagnostic data.
     """
+    from iris.cube import CubeList
+
     from improver.wind_calculations.wind_gust_diagnostic import WindGustDiagnostic
 
-    result = WindGustDiagnostic(wind_gust_percentile, wind_speed_percentile)(
+    wind_gust, wind_speed = CubeList(cubes).extract(
+        ["wind_speed_of_gust", "wind_speed"]
+    )
+
+    return WindGustDiagnostic(wind_gust_percentile, wind_speed_percentile)(
```

Comment:
I think that the WindGustDiagnostic plugin could also be updated with the usage of the    `assert_spatial_coords_match` and `assert_time_coords_valid` functions. I can see that this plugin is already comparing whether the time coordinates are valid on [these lines](https://github.com/metoppv/improver/blob/master/improver/wind_calculations/wind_gust_diagnostic.py#L185-L202) by checking whether the wind speed time point is within the time bounds provided by the wind gust cube. This kind of check isn't supported by `assert_time_coords_valid` but we could consider whether `assert_time_coords_valid` could be updated to support this kind of check.